compiler/rustc_trait_selection/src/error_reporting/traits/ambiguity.rs

```diff
@@ -610,6 +610,24 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
                 )
                 .with_span_label(span, format!("cannot normalize `{alias}`"))
             }
+            ty::PredicateKind::Clause(ty::ClauseKind::UnstableFeature(sym)) => {
+                if let Some(e) = self.tainted_by_errors() {
+                    return e;
+                }
+
+                let mut err = self.dcx().struct_span_err(
+                    span,
+                    format!("unstable feature `{sym}` is used without being enabled."),
```

Comment:
I would try to mirror the existing errors for unstable library features if `staged_api` is not enabled

```
error[E0658]: use of unstable library feature `slice_partition_dedup`
 --> src/main.rs:3:7
  |
3 |     x.partition_dedup();
  |       ^^^^^^^^^^^^^^^
  |
  = note: see issue #54279 <https://github.com/rust-lang/rust/issues/54279> for more information
  = help: add `#![feature(slice_partition_dedup)]` to the crate attributes to enable
  = note: this compiler was built on 2025-07-02; consider upgrading it if it is out of date
```
not sure whehter you could even reuse the reporting machinery we already use for methods etc